src/profiler.cpp

```diff
@@ -63,6 +64,20 @@ static J9WallClock j9_wall_clock;
 static ITimer itimer;
 static Instrument instrument;
 
+static pthread_key_t local_context_key;
+
+static void destroyContext(void* value) {
+    if (value) {
+        free(value);
+    }
+}
+
+static void* createContext() {
+    (void) pthread_key_create(&local_context_key, destroyContext);
+    return NULL;
+}
+
+static void* init_local_context = createContext();
```

Comment:
`__attribute__((constructor))` will make the intention clearer and will also avoid the unused variable
(but that's just a question of personal preferences)